File: libs/eave-stdlib-py/src/eave/stdlib/core_api/models/atlassian.py
import uuid
from typing import List, Optional
from eave.stdlib.core_api.models import BaseInputModel, BaseResponseModel


class ConfluenceSpace(BaseResponseModel):
    key: str
    name: str


class AtlassianInstallation(BaseResponseModel):
    id: uuid.UUID
    team_id: uuid.UUID
    """eave TeamOrm model id"""
    atlassian_cloud_id: str
    confluence_space_key: Optional[str] = None
    "DEPRECATED: Always None"
    available_confluence_spaces: Optional[List[ConfluenceSpace]]


AtlassianInstallationPeek = AtlassianInstallation
"""Type Alias for naming consistency with other integrations"""


class AtlassianInstallationInput(BaseInputModel):
    atlassian_cloud_id: str

File: libs/eave-stdlib-ts/src/core-api/models/atlassian.ts
export type ConfluenceSpace = {
  key: string;
  name: string;
};

export type AtlassianInstallation = {
  id: string;
  /** eave TeamOrm model id */
  team_id: string;
  atlassian_cloud_id: string;
  confluence_space: string | null;
  available_confluence_spaces: Array<ConfluenceSpace> | null;
};

// Typealias for naming consistency
export type AtlassianInstallationPeek = AtlassianInstallation;
